### ISSUE TITLE
Model fails to save when using callbacks

### ISSUE THREAD

**Issue by Nelson-Gon** [Nelson-Gon/cytounet#27]
- [x] I have read the [wiki](https://github.com/Nelson-Gon/cytounet/wiki) and my issue is not 
handled (sufficiently) there or is not answered at all.

**Describe the bug**

After training, I used a callback as shown in the example [notebook](https://github.com/Nelson-Gon/cytounet/blob/4441d83d7a850695dc94b2f99725a4e4d07d7c59/examples/example_usage.ipynb) and found that the model was not saved at the end of each epoch. 

**To Reproduce**

Run the exact code in the above notebook. 

**Expected behavior**

I expected to save the model at the end of each epoch based on the loss' improvement(or not). 

**Unexpected behavior**

I got no error message. The model trained successfully but required that I manually save the model at the end of the entire training. 

**System Details**

Using developer version with this [hash](https://github.com/Nelson-Gon/cytounet/tree/4441d83d7a850695dc94b2f99725a4e4d07d7c59).

**Comment by Nelson-Gon** [comment]
When training, manually save or provide  a callback. Otherwise, use `finetune` instead of `train`. The former saves at the end of each epoch.
